src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystemProvider.java

```diff
@@ -284,6 +293,15 @@ public OutputStream newOutputStream(Path path, OpenOption... options)
         return toZipPath(path).readAttributes(attributes, options);
     }
 
+    @Override
+    public <A extends BasicFileAttributes> A readAttributesIfExists(Path path,
+                                                                    Class<A> type,
+                                                                    LinkOption... options)
+        throws IOException
+    {
+        return exists(path) ? readAttributes(path, type, options) : null;
```

Comment:
If you add the following to ZipPath then it would allow you to implement it in one step rather than two:

```
ZipFileAttributes readAttributesIfExits() throws IOException {
    return zfs.getFileAttributes(getResolvedPath());
 }
```